docs/src/examples/NPG.jl

```diff
@@ -0,0 +1,171 @@
+    policy,
+    value,
+    valuetrainer;
+    gamma = 0.995,
+    gaelambda = 0.97,
+    norm_step_size = 0.05,
+    Hmax = 1000,
+    N = 10000,
+);
+
+#md # ## Running Experiments
+# Finally, let's spin on our iterator 200 times, plotting every 20 iterations.
+# This lets us break out of the loop if certain conditions are met, or re-start training
+# manually if needed. We of course wish to track results, so we create an `Experiment` to
+# which we can save data. We also have useful timing information displayed every
+# 20 iterations to understand CPU performance.
+exper = Experiment("/tmp/hopper_example.jlso", overwrite = true)
+lg = ULogger() # walks, talks, and acts like a Julia logger. See the docs on UniversalLogger for more info.
+for (i, state) in enumerate(npg)
+    if i > 200
+        ## serialize some stuff and quit
+        exper[:policy] = npg.policy
+        exper[:value] = npg.value
+        exper[:etype] = LyceumMuJoCo.HopperV2
+        exper[:meanstates] = state.meanbatch
+        exper[:stocstates] = state.stocbatch
+        break
+    end
+
+    ## log everything in `state` except meanbatch and stocbatch
+    push!(lg, :algstate, filter_nt(state, exclude = (:meanbatch, :stocbatch)))
+
+    if mod(i, 20) == 0
+        x = lg[:algstate]
+        ## The following are helper functions for plotting to the terminal.
+        ## The first plot displays the eval function for our stochastic
+        ## and mean policy rollouts.
+        display(expplot(
+            Line(x[:stocterminal_eval], "StocLastE"),
+            Line(x[:meanterminal_eval], "MeanLastE"),
+            title = "Evaluation Score, Iter=$i",
+            width = 60,
+            height = 8,
+        ))
+        ## While the second one similarly plots the reward.
+        display(expplot(
+            Line(x[:stoctraj_reward], "StocR"),
+            Line(x[:meantraj_reward], "MeanR"),
+            title = "Reward, Iter=$i",
+            width = 60,
+            height = 8,
+        ))
+
+        ## The following are timing values for various parts of the Natural Policy Gradient
+        ## algorithm at the last iteration, useful for finding performance bottlenecks in the algorithm.
+        println("elapsed_sampled  = ", state.elapsed_sampled)
+        println("elapsed_gradll   = ", state.elapsed_gradll)
+        println("elapsed_vpg      = ", state.elapsed_vpg)
+        println("elapsed_cg       = ", state.elapsed_cg)
+        println("elapsed_valuefit = ", state.elapsed_valuefit)
+    end
+end
+
+# Let's go ahead and plot the final reward trajectory for our stochastic and mean policies
+# to see how we did.
+plot!(
+    plot(lg[:algstate][:meantraj_reward], label = "Mean Policy", title = "HopperV2 Reward"),
+    lg[:algstate][:stoctraj_reward],
+    label = "Stochastic Policy",
+)
+
+# and save the logged results to the Experiment
+for (k, v) in get(lg)
+    exper[k] = v
+end
+finish!(exper); # flushes everything to disk
```

Comment:
can you add a test for NPG similar to MPPI?

Reply:
No, as it's too noisy to add any meaningful test.

Reply:
added tests